### PR TITLE
[sfputil] Add support of platform.json

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

> This PR is dependent on [sonic-platform-common/pull/72](https://github.com/Azure/sonic-platform-common/pull/72)

All three PRs are necessary to run `show interfaces transceiver` command.

1. [Azure/sonic-buildimage#3912](https://github.com/Azure/sonic-buildimage/pull/3912)
2. [Azure/sonic-platform-common#72](https://github.com/Azure/sonic-platform-common/pull/72)
3. [Azure/sonic-utilities#767](https://github.com/Azure/sonic-utilities/pull/767)

**- What I did**
Add support of platform.json in sfputil to get correct output of `show interfaces transceiver`

**- How to verify it**
Check whether all the below-mentioned CLI's are working correctly.
```
Usage: show interfaces transceiver [OPTIONS] COMMAND [ARGS]...

  Show SFP Transceiver information

Options:
  -?, -h, --help  Show this message and exit.

Commands:
  eeprom    Show interface transceiver EEPROM information
  lpmode    Show interface transceiver low-power mode status
  presence  Show interface transceiver presence
```

### CODE REVIEW

File: sfputil/main.py [hunk 1]
```diff
@@ -329,14 +331,16 @@ def get_path_to_port_config_file():
...
     # First check for the presence of the new 'port_config.ini' file
-    port_config_file_path = "/".join([hwsku_path, "port_config.ini"])
+    port_config_file_path = "/".join([platform_path, PLATFORM_JSON])
+    if not os.path.isfile(port_config_file_path):
+        # platform.json doesn't exist. Try loading the legacy 'port_config.ini' file
+        port_config_file_path = "/".join([hwsku_path, PORT_CONFIG_INI])
     if not os.path.isfile(port_config_file_path):
         # port_config.ini doesn't exist. Try loading the legacy 'portmap.ini' file
-        port_config_file_path = "/".join([hwsku_path, "portmap.ini"])
+        port_config_file_path = "/".join([hwsku_path, PORTMAP_INI])
```
Comment (jleveque) [inline sfputil/main.py]: `portmap.ini` is legacy, and was deprecated years ago in favor of `port_config.ini`. I think at this point, we can remove reference to this file entirely.

@lguohan: Do you agree?

Comment (jleveque) [reply sfputil/main.py:340]: @samaity: Please remove the `portmap.ini` references as part of this PR.

Comment (samaity) [reply sfputil/main.py]: #done

**Comment by lguohan** [comment]
retest this please

**Pull request merged**